### DISCUSSION TITLE
Run M-PACE case and change vertical levels

### DISCUSSION THREAD

**Discussion by nickszap** [NCAR/ccpp-scm#383] [Cases, Initial Conditions, Forcing]
Hi, I am trying to follow the current instructions to experiment with the M-PACE case in DEPHY format by changing the number of vertical levels and have two hopefully quick questions.

Q1) First, when I try to run a DEPHY case, I get an error:
"There was an error reading the namelist case_config in the file input_experiment.nml; error code = 19" 
Does this work for you?

Q2) Then, I am a bit confused with the vert_coord_file.
ccpp-scm/scm/src/scm_vgrid.F90 shows that ak bk should go surface to TOA and are then flipped so that 
scm_state%pres_i(:,k) = scm_state%a_k(k) + scm_state%b_k(k)*p_ref.
All of the files in ccpp-scm/scm/data/vert_coord_data/ look like
0 1
…
0 0
However, all of the levels in ccpp-scm/scm/src/fv_eta.h look like
a0 0
…
0 1
with a0>0. Does this a0 need to match ptop?

I very much appreciate your help.
Cheers,
Nick

**Comment by grantfirl** [comment]
Hi Nick,

I'll answer Q2. 

The files in ccpp-scm/scm/data/vert_coord_data/ were taken from the GSM-based GFS several years ago. Yes, their data go from surface to TOA. This data is vertically flipped immediately after being read in scm_input.F90 to match what is in fv3_eta.h, so that it is going from TOA to surface.

At the end of the `get_FV3_vgrid()` subroutine, ak/bk are flipped from TOA-to-surface to surface-to-TOA. This results in one flip for data coming from fv3_eta.h and two flips for data coming from a file in vert_coord_data.

I don't know if it is required that a0 needs to match ptop or not. That is probably a question for the FV3 dycore folks since fv3_eta.h is taken directly from the FV3 repo.

**Comment by hertneky** [comment]
For Q1, are you testing the MPACE/REF case? I received the same error when trying to run that case as is.
In the case_config/MPACE_REF.nml, the line causing the troubles is reference_profile_dir = '../data/processed_case_input'. I was able to get the case to run by removing that config option. It defaults to the 'scm/data/processed_case_input' directory if you don't have that set, so it's not needed. In the meantime, I will look more into what is going on with that config option.

**Reply by hertneky** [reply]
Nick - Just to update, the config option reference_profile_dir is not used and we have a PR to remove that from all case configs. In addition, there is a also a fix for a separate issue that causes the SCM to stop if surfaceForcing isn't set to LSM (even though it runs to completion). I will let you know when that is available.

**Reply by hertneky** [reply]
Nick - #384 was merged to the main repo. The SCM DEPHY cases, including MPACE should work with that latest update.

**Comment by nickszap** [comment]
Thank you so much for your informative responses and solutions, @grantfirl and @hertneky! 
I can now run the MPACE/REF case without any changes.

Maybe I should mention that physics suites GFS_v16 and GFS_v17_p8 run but the rest using MYNN-EDMF (RAP, RRFS_v1beta, WoFS, & HRRR) fail with segfault in module_sf_mynn.F90.
